### PR TITLE
huobi - new endpoints

### PR DESCRIPTION
**Pull request by ttodua**
updated per announcement - https://www.huobi.com/support/en-us/detail/94916436903925

### CODE REVIEW

File: js/huobi.js [hunk 1]
```diff
@@ -239,18 +239,6 @@ module.exports = class huobi extends Exchange {
...
-                        'detail/merged': 1, // 获取聚合行情(Ticker)
-                        'depth': 1, // 获取 Market Depth 数据
-                        'trade': 1, // 获取 Trade Detail 数据
-                        'history/trade': 1, // 批量获取最近的交易记录
-                        'detail': 1, // 获取 Market Detail 24小时成交量数据
-                        'tickers': 1,
-                        'etp': 1, // 获取杠杆ETP实时净值
-                    },
-                },
                 'public': {
```
Comment (ttodua) [inline js/huobi.js:242]: I can say with my responsibility, that these `market` endpoints were just of no use. all of them were for spot, and duplicated. they are defined already in `spot > public > get > `  'market/whatever'  endpoints, and used the same url in the end. So, we have to totally remove `market` tree, they are also referenced nowhere.

File: js/huobi.js [hunk 2]
```diff
@@ -5511,9 +5511,7 @@ module.exports = class huobi extends Exchange {
         const query = this.omit (params, this.extractParams (path));
         if (typeof api === 'string') {
             // signing implementation for the old endpoints
-            if (api === 'market') {
-                url += api;
-            } else if ((api === 'public') || (api === 'private')) {
+            if ((api === 'public') || (api === 'private')) {
```
Comment (ttodua) [inline js/huobi.js:5514]: here removed `api === market` block complerely.

**Comment by ttodua** [comment]
@carlosmiei current PR is ready. however, if you can wait one more ~12 hrs, I will have updated implementations too (if there are any which needs update)

**Comment by ttodua** [comment]
@carlosmiei now, I've finished updating implementations too. I've tested those new endpoints with new arguments, and made them work as expected. the funding history endpoints were also updated and made compatible with new endpoints (tested) , however, as I don't have any recent funding history, the returned array from raw response is empty, and couldnt see the response myself, however, I am confident that (per know indicators) that the implementation should be correct (as down the road, when i made request with incorrect arguments, endpoint responded with related exceptions, and now it returns expected response structure, ,even though it's empty).

File: js/huobi.js [hunk 3]
```diff
@@ -5683,43 +5689,42 @@ module.exports = class huobi extends Exchange {
...
+                //                 "margin_account": "USDT",
+                //                 "face_margin_account": ""
+                //             }
+                //         ],
+                //         "ts": 1604312615051
+                //     }
+                //
             } else {
-                throw new ExchangeError (this.id + ' fetchFundingHistory() only makes sense for swap contracts');
+                method = 'contractPrivatePostApiV3ContractFinancialRecordExact';
```
Comment (ttodua) [inline js/huobi.js:5721]: Idk why there was exception before, and why it doesn't make sense to have funding history for `inverse futures` ?  it also has it's own endpoint, and the concept shouldnt be any different from `inverse swap`, as funding history applies to both of them.

Comment (carlosmiei) [reply js/huobi.js:5721]: hum, so probably was a bug

**Comment by carlosmiei** [comment]
LGTM ✅ 

```
p huobi fetchMyTrades "LTC/USDT"               
Python v3.9.7
CCXT v1.94.2
huobi.fetchMyTrades(LTC/USDT)
[{'amount': 0.1622,
  'cost': 8.56416,
  'datetime': '2022-10-12T09:46:39.820Z',
  'fee': {'cost': 0.01712832, 'currency': 'USDT'},
  'fees': [{'cost': 0.01712832, 'currency': 'USDT'}],
  'id': '630793863411068',
  'info': {'created-at': '1665567999820',
           'fee-currency': 'usdt',
           'fee-deduct-currency': '',
           'fee-deduct-state': 'done',
           'filled-amount': '0.1622',
           'filled-fees': '0.01712832',
           'filled-points': '0.0',
           'id': '630793863411068',
           'match-id': '113526568872',
           'order-id': '651351299131430',
           'price': '52.8',
           'role': 'taker',
           'source': 'spot-web',
           'symbol': 'ltcusdt',
           'trade-id': '101547647943',
           'type': 'sell-limit'},
  'order': '651351299131430',
  'price': 52.8,
  'side': 'sell',
  'symbol': 'LTC/USDT',
  'takerOrMaker': 'taker',
  'timestamp': 1665567999820,
  'type': 'limit'}]
```

```
p huobi fetchMyTrades "BTC/USDT:USDT" None 1   
Python v3.9.7
CCXT v1.94.2
huobi.fetchMyTrades(BTC/USDT:USDT,None,1)
[{'amount': 1.0,
  'cost': 19.0481,
  'datetime': '2022-10-12T02:39:00.344Z',
  'fee': {'cost': -0.00952405, 'currency': 'USDT'},
  'fees': [{'cost': -0.00952405, 'currency': 'USDT'}],
  'id': '118427751608-1029704737200709632-1',
  'info': {'business_type': 'swap',
           'contract_code': 'BTC-USDT',
           'contract_type': 'swap',
           'create_date': '1665542340344',
           'direction': 'sell',
           'fee_asset': 'USDT',
           'id': '118427751608-1029704737200709632-1',
           'margin_account': 'USDT',
           'margin_mode': 'cross',
           'match_id': '118427751608',
           'offset': 'close',
           'offset_profitloss': '-0.009300000000000000',
           'order_id': '1029704737200709632',
           'order_id_str': '1029704737200709632',
           'order_source': 'api',
           'pair': 'BTC-USDT',
           'query_id': '749816820',
           'real_profit': '-0.009300000000000000',
           'reduce_only': '1',
           'role': 'Taker',
           'symbol': 'BTC',
           'trade_fee': '-0.009524050000000000',
           'trade_partition': 'USDT',
           'trade_price': '19048.100000000000000000',
           'trade_turnover': '19.048100000000000000',
           'trade_volume': '1.000000000000000000'},
  'order': '1029704737200709632',
  'price': 19048.1,
  'side': 'sell',
  'symbol': 'BTC/USDT:USDT',
  'takerOrMaker': 'taker',
  'timestamp': 1665542340344,
  'type': None}]
```

```
 p huobi fetchMyTrades "LTC/USD:LTC" None 1 '{"marginMode":"isolated"}'          
Python v3.9.7
CCXT v1.94.2
huobi.fetchMyTrades(LTC/USD:LTC,None,1,{'marginMode': 'isolated'})
[{'amount': 2.0,
  'cost': 20.0,
  'datetime': '2022-10-12T09:55:38.409Z',
  'fee': {'cost': -0.00018986140117714, 'currency': 'LTC'},
  'fees': [{'cost': -0.00018986140117714, 'currency': 'LTC'}],
  'id': '98664750453-1029814619839365120-1',
  'info': {'contract_code': 'LTC-USD',
           'create_date': '1665568538409',
           'direction': 'sell',
           'fee_asset': 'LTC',
           'id': '98664750453-1029814619839365120-1',
           'match_id': '98664750453',
           'offset': 'close',
           'offset_profitloss': '-0.000072081017910840',
           'order_id': '1029814619839365120',
           'order_id_str': '1029814619839365120',
           'order_source': 'web',
           'query_id': '995298518',
           'real_profit': '-0.000072081017910840',
           'role': 'Taker',
           'symbol': 'LTC',
           'trade_fee': '-0.000189861401177140',
           'trade_price': '52.670000000000000000',
           'trade_turnover': '20.000000000000000000',
           'trade_volume': '2.000000000000000000'},
  'order': '1029814619839365120',
  'price': 52.67,
  'side': 'sell',
  'symbol': 'LTC/USD:LTC',
  'takerOrMaker': 'taker',
  'timestamp': 1665568538409,
  'type': None}]
  ```
  

```
 p huobi fetchOrders "LTC/USDT"
Python v3.9.7
CCXT v1.94.2
huobi.fetchOrders(LTC/USDT)
[{'amount': 0.1622,
  'average': 52.8,
  'clientOrderId': None,
  'cost': 8.56416,
  'datetime': '2022-10-12T09:46:39.804Z',
  'fee': {'cost': 0.01712832, 'currency': 'USDT'},
  'fees': [{'cost': 0.01712832, 'currency': 'USDT'}],
  'filled': 0.1622,
  'id': '651351299131430',
  'info': {'account-id': '44234548',
           'amount': '0.162200000000000000',
           'canceled-at': '0',
           'client-order-id': '',
           'created-at': '1665567999804',
           'field-amount': '0.162200000000000000',
           'field-cash-amount': '8.564160000000000000',
           'field-fees': '0.017128320000000000',
           'finished-at': '1665567999823',
           'id': '651351299131430',
           'price': '52.800000000000000000',
           'source': 'spot-web',
           'state': 'filled',
           'symbol': 'ltcusdt',
           'type': 'sell-limit'},
  'lastTradeTimestamp': None,
  'postOnly': None,
  'price': 52.8,
  'remaining': 0.0,
  'side': 'sell',
  'status': 'closed',
  'stopPrice': None,
  'symbol': 'LTC/USDT',
  'timeInForce': None,
  'timestamp': 1665567999804,
  'trades': [],
  'type': 'limit'}]
  ```

```
p huobi fetchClosedOrders "BTC/USDT:USDT" None 1     
Python v3.9.7
CCXT v1.94.2
huobi.fetchClosedOrders(BTC/USDT:USDT,None,1)
[{'amount': 1.0,
  'average': 19048.1,
  'clientOrderId': None,
  'cost': 19.0481,
  'datetime': '2022-10-12T02:39:00.333Z',
  'fee': {'cost': -0.00952405, 'currency': 'USDT'},
  'fees': [{'cost': -0.00952405, 'currency': 'USDT'}],
  'filled': 1.0,
  'id': '1029704737200709632',
  'info': {'business_type': 'swap',
           'contract_code': 'BTC-USDT',
           'contract_type': 'swap',
           'create_date': '1665542340333',
           'direction': 'sell',
           'fee': '-0.009524050000000000',
           'fee_asset': 'USDT',
           'is_tpsl': '0',
           'lever_rate': '1',
           'liquidation_type': '0',
           'margin_account': 'USDT',
           'margin_asset': 'USDT',
           'margin_frozen': '0E-18',
           'margin_mode': 'cross',
           'offset': 'close',
           'order_id': '1029704737200709632',
           'order_id_str': '1029704737200709632',
           'order_price_type': '1',
           'order_source': 'api',
           'order_type': '1',
           'pair': 'BTC-USDT',
           'price': '19048.000000000000000000',
           'profit': '-0.009300000000000000',
           'query_id': '39443371865',
           'real_profit': '-0.0093',
           'reduce_only': '1',
           'status': '6',
           'symbol': 'BTC',
           'trade_avg_price': '19048.1',
           'trade_partition': 'USDT',
           'trade_turnover': '19.048100000000000000',
           'trade_volume': '1.000000000000000000',
           'update_time': '1665542341044',
           'volume': '1.000000000000000000'},
  'lastTradeTimestamp': None,
  'postOnly': None,
  'price': 19048.0,
  'remaining': 0.0,
  'side': 'sell',
  'status': 'closed',
  'stopPrice': None,
  'symbol': 'BTC/USDT:USDT',
  'timeInForce': None,
  'timestamp': 1665542340333,
  'trades': [],
  'type': '1'}]
```

**Review by carlosmiei — APPROVED** [review]
